compiler/luci/pass/src/ReplaceMulAddWithDepthwiseConvPass.cpp

```diff
@@ -28,11 +28,10 @@ luci::CircleConst *create_weights_from_gamma(luci::CircleConst *gamma)
 {
   assert(gamma->rank() == 1 or gamma->rank() == 4);
 
-  uint32_t channel_size = 0;
-  if (gamma->rank() == 1)
-    channel_size = gamma->dim(0).value();
-  else if (gamma->rank() == 4)
-    channel_size = gamma->dim(3).value();
+  uint32_t channel_idx = gamma->rank() - 1;
+  uint32_t channel_size = gamma->dim(channel_idx).value();
+  for (uint32_t i = 0; i < channel_idx; i++)
+    assert(gamma->dim(i).value() == 1); // FIX is_batchnorm_mul UNLESS
```

Comment:
Could you explain the intention of this for loop?

Reply:
The intention is to check whether gamma/beta's shape is broadcastable in the channel direction.

For example, if channel size is 16, gamma/beta's shape can be {16}, {1, 16}, {1, 1, 16}, {1,1,1,16}.